### DISCUSSION TITLE
Update from 3.0 to 4.5 - config backend

### DISCUSSION THREAD

**Discussion by meszat2** [OpenIdentityPlatform/OpenDJ#246] [Q&A]
Hi!

I have a problem with the upgrade (from 3.0 to 4.5).
I have made full backup.
`./backup --backupAll --backupDirectory /home/opendj/backup`
I have tried to restore config backend.
`./restore --backupDirectory /home/opendj/backup/config --offline`

I have received error message:
[29/Jul/2022:12:17:52 +0200] category=TOOLS severity=ERROR msgID=315 msg=The backups contained in directory /home/opendj/backup/config were taken from a Directory Server backend defined in configuration entry ds-cfg-backend-id=config,cn=Backends,cn=config but no such backend is available

I have found this record in backup:
```
dn: ds-cfg-backend-id=config,cn=Backends,cn=config
objectClass: top
objectClass: ds-cfg-config-file-handler-backend
objectClass: ds-cfg-backend
ds-cfg-enabled: true
ds-cfg-base-dn: cn=config
ds-cfg-backend-id: config
ds-cfg-java-class: org.opends.server.extensions.ConfigFileHandler
ds-cfg-writability-mode: enabled
```

I should restore configurations, because it contains the following informations:
- password policies,
- indexes,
- server settings,
- smtp settings,
- log settings,
- ACIs

How can I restore config backend?

Regards,
Attila

**Comment by vharseko** [comment]
https://github.com/OpenIdentityPlatform/OpenDJ/wiki/How-To#backup-and-restore

```shell
./restore --backupDirectory /home/opendj/backup/config/userRoot/  --offline
```
